File: platform/sdk/nodejs/examples/t3-04-abort.js
const { CloudClient } = require("../xuanji-sdk-cloud");

function main() {
  const client = new CloudClient();
  const bucket = "abkt";
  const key = "abort/file.tmp";
  const r0 = client.createMultipartUpload(bucket, key);
  if (!r0.ok) throw new Error("create failed");
  const uid = r0.upload_id;
  client.uploadPart(bucket, key, uid, 1, "partial");
  const before = client.listMultipartUploads();
  if (before.count !== 1) throw new Error("expected 1 before abort, got " + before.count);
  const a = client.abortMultipartUpload(uid);
  if (!a.ok || !a.aborted) throw new Error("abort should return ok:true aborted:true");
  const after = client.listMultipartUploads();
  if (after.count !== 0) throw new Error("expected 0 after abort, got " + after.count);
  console.log("XJ-OK: t3-04-abort removed=" + uid + " remaining=" + after.count);
}

if (require.main === module) main();
module.exports = main;

File: platform/sdk/python/examples/t3_04_abort.py
import sys, os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), ".."))
from mox_sdk_cloud import CloudClient

def main():
    client = CloudClient()
    bucket = "abkt"
    key = "abort/file.tmp"
    r0 = client.createMultipartUpload(bucket, key)
    if not r0["ok"]:
        raise RuntimeError("create failed")
    uid = r0["upload_id"]
    client.uploadPart(bucket, key, uid, 1, "partial")
    before = client.listMultipartUploads()
    if before["count"] != 1:
        raise RuntimeError(f"expected 1 before abort got {before['count']}")
    a = client.abortMultipartUpload(uid)
    if not a["ok"] or not a["aborted"]:
        raise RuntimeError("abort should be ok+aborted")
    after = client.listMultipartUploads()
    if after["count"] != 0:
        raise RuntimeError(f"expected 0 after abort got {after['count']}")
    print(f"XJ-OK: t3_04_abort removed={uid} remaining={after['count']}")

if __name__ == "__main__":
    main()
